rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -138,7 +137,22 @@ namespace rsx
 				glPixelStorei(GL_UNPACK_SWAP_BYTES, GL_TRUE);
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
 
-				// TODO: texture swizzling
+				if (is_swizzled)
+				{
+					u16 height = tex.height();
+					u16 width = tex.width();
+
+					std::vector<u16> swz_buf(width * height);
+					u16 *src = (u16*)pixels;
```

Comment:
rsx::convert_linear_swizzle receives void* pointers and not const void*. I could rewrite the function to accept const void* but I'm trying to keep the changeset as small as possible.